flink-runtime/src/main/java/org/apache/flink/runtime/jobgraph/JobVertex.java

```diff
@@ -308,7 +308,7 @@ public int getParallelism() {
      * @param parallelism The parallelism for the task.
      */
     public void setParallelism(int parallelism) {
-        if (parallelism < 1) {
+        if (parallelism < 1 && parallelism != ExecutionConfig.PARALLELISM_DEFAULT) {
```

Comment:
Will we pass `ExecutionConfig.PARALLELISM_DEFAULT` here ?

Reply:
It will only happen in tests at the moment. However, I feel not to rework the test utils because I think PARALLELISM_DEFAULT should be a valid value. It is already accepted in many places and I prefer them to be aligned.